### PR TITLE
qwen3-14b decode_layer: self-contained random-golden unit test + --max-seq

### PR DESCRIPTION
**Pull request by lwDavid**
## Summary

Rework the single-layer `__main__` unit test in `models/qwen3/14b/decode_layer.py` so it no longer depends on pre-generated golden files (the referenced `gen_e2e_golden.py` is not in the repo). The test now generates a deterministic **random** fixture, computes the golden **on the fly** with a torch reference, runs `qwen3_decode_mpmd` on device through the `golden/` harness, and validates the output.

## What changed

- **Random fixture + on-the-fly golden**, via the `golden/` harness:
  - `random_inputs(full_seq, seed)` — deterministic random inputs.
  - `golden_decode_layer(values)` — torch reference mirroring the kernel math **and its bf16 cast points** (RMSNorm → QKV proj → per-head QK-norm → RoPE → KV-cache write at `pos=seq_len-1` → GQA flash attention → out_proj → residual → post-RMSNorm → SwiGLU MLP → residual).
  - Driven by `golden.run_jit(..., golden_fn=golden_decode_layer, compare_fn={"out": ratio_allclose(atol=rtol=3e-3, max_error_ratio=0.02)})`.
- **`--max-seq`**: set every `seq_len` to `MAX_SEQ` (full KV cache) for a stable, maximum-load performance run (pairs well with `--enable-l2-swimlane`). Default samples varied random lengths, which also exercises the ragged load-balancer path.
- **`--seed`**: reproducible fixtures.
- `*sim` platforms and `--smoke` stay **compile-only** — unchanged CI sweep behavior.
- `--validate-fwd` unchanged.

Fixture scales keep the unnormalized residual-stream output `O(1)` and attention well conditioned (small past KV + nonzero-mean V), so the device bf16 output matches the fp32 reference within the `ratio_allclose` tolerance.

## Testing

On Ascend 910B (a2a3):

```
# default (varied random lengths)
python models/qwen3/14b/decode_layer.py -p a2a3 -d <dev>
# -> 'out' PASS  ratio_allclose(atol=0.003, rtol=0.003, max_error_ratio=0.02)

# stable max-load perf + swimlane
python models/qwen3/14b/decode_layer.py -p a2a3 -d <dev> --max-seq --enable-l2-swimlane
# -> 'out' PASS  + swimlane JSON written
```

Both pass. Compile-only smoke (`-p a2a3sim`) and ruff / header / english-only lint checks also pass.

### CODE REVIEW

File: models/qwen3/14b/decode_layer.py [hunk 1]
```diff
@@ -1162,7 +1162,175 @@ def _backend_type(platform: str) -> BackendType:
...
+
+
+def _build_specs(inputs: dict, TensorSpec) -> list:
+    """TensorSpec list in qwen3_decode_mpmd parameter order, from a fixture dict."""
+    specs = [
+        TensorSpec(name, list(inputs[name].shape), inputs[name].dtype, init_value=inputs[name])
+        for name in INPUT_NAMES
+    ]
+    specs.append(TensorSpec("out", [BATCH, HIDDEN], torch.bfloat16, is_output=True))
+    return specs
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply models/qwen3/14b/decode_layer.py]: Fixed in 2c1ec14 — `TensorSpec` is now imported inside `_build_specs` and the parameter is dropped. I kept `torch` as the existing module-level import (it's used throughout the file), so I didn't add the redundant local `import torch`.

File: models/qwen3/14b/decode_layer.py [hunk 2]
```diff
@@ -1188,35 +1363,59 @@ def _backend_type(platform: str) -> BackendType:
...
         raise SystemExit(0)
 
-    backend_type = _backend_type(args.platform)
+    # ── Default single-layer unit test: RANDOM inputs, on-the-fly torch golden,
+    # on-device run + compare, all through the golden/ harness. ──
+    if not args.validate_fwd:
+        from golden import TensorSpec, ratio_allclose, run_jit
+
+        inputs = random_inputs(full_seq=args.max_seq, seed=args.seed)
+        specs = _build_specs(inputs, TensorSpec)
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply models/qwen3/14b/decode_layer.py]: Fixed in 2c1ec14 — dropped the `TensorSpec` argument at the call site and removed it from the `__main__` `from golden import ...` list.
